swift/ql/lib/codeql/swift/dataflow/internal/FlowSummaryImplSpecific.qll

```diff
@@ -32,29 +32,63 @@ ArgumentPosition callbackSelfParameterPosition() { result instanceof ThisArgumen
 SummaryCall summaryDataFlowCall(SummaryNode receiver) { receiver = result.getReceiver() }
 
 /** Gets the type of content `c`. */
-DataFlowType getContentType(ContentSet c) { any() }
+DataFlowType getContentType(ContentSet c) {
+  exists(Content cc |
+    c.isSingleton(cc) and
+    result = cc.getType()
+  )
+}
 
 /** Gets the type of the parameter at the given position. */
-DataFlowType getParameterType(SummarizedCallable c, ParameterPosition pos) { any() }
+DataFlowType getParameterType(SummarizedCallable c, ParameterPosition pos) {
+  result = getDataFlowType(c.getParam(pos.(PositionalParameterPosition).getIndex()).getType())
+  or
+  pos instanceof ThisParameterPosition and
+  result = getDataFlowType(c.getSelfParam().getType())
+}
 
 /** Gets the return type of kind `rk` for callable `c`. */
 bindingset[c]
-DataFlowType getReturnType(SummarizedCallable c, ReturnKind rk) { any() }
+DataFlowType getReturnType(SummarizedCallable c, ReturnKind rk) {
+  rk instanceof NormalReturnKind and
+  result = getDataFlowType(c.getResultType())
+  or
+  exists(ParamDecl p |
+    p = c.getParam(rk.(ParamReturnKind).getIndex()) and
+    p.isInout() and
+    result = getDataFlowType(p.getType())
+  )
+  or
+  rk.(ParamReturnKind).getIndex() = -1 and
+  c.getSelfParam().isInout() and
+  result = getDataFlowType(c.getSelfParam().getType())
+}
 
 /**
  * Gets the type of the parameter matching arguments at position `pos` in a
  * synthesized call that targets a callback of type `t`.
  */
-DataFlowType getCallbackParameterType(DataFlowType t, ArgumentPosition pos) { any() }
+DataFlowType getCallbackParameterType(DataFlowType t, ArgumentPosition pos) {
+  exists(t) and
+  exists(pos) and
+  result.asType() instanceof AnyType
+}
 
 /**
  * Gets the return type of kind `rk` in a synthesized call that targets a
  * callback of type `t`.
  */
-DataFlowType getCallbackReturnType(DataFlowType t, ReturnKind rk) { any() }
+DataFlowType getCallbackReturnType(DataFlowType t, ReturnKind rk) {
+  exists(t) and
+  exists(rk) and
+  result.asType() instanceof AnyType
+}
```

Comment:
Same here. This looks like a CP between all types and all return kinds. I guess this is fine if the `getCallbackParameterType` case is also fine?

Reply:
It was fine but we don't actually need these anymore - they were part of the old API but not the new one.